### ISSUE TITLE
mat-menu trigger is setting aria-expanded to null instead of false so screenreader does not announce collapse

### ISSUE THREAD

**Issue by Priscilla-developer** [angular/components#23594]
#### Reproduction

Use StackBlitz to reproduce your issue: 
<button mat-icon-button
        #menuButton
        [matMenuTriggerFor]="exampleMenu" 
        aria-expanded = "false"
        aria-haspopup = "true"
>
<mat-icon fontSet = "fa" fontIcon = "fa-ellipsis-v"></mat-icon>
</button>

<mat-menu #exampleMenu="matMenu">

Steps to reproduce:
1. Add a button with [matTriggerFor] for a mat menu

2. Inspect the element in the browser

3. Click on the button to open the menu- it sets the aria-expanded to true

4. Inspect the element when the menu is closed - having it closed does not set it to false (sets it to null)
 
 
#### Expected Behavior

When the menu is closed, aria-expanded should be false so screen reader will read as collapse

#### Actual Behavior

aria-expanded is set to null and the screen reader is not stating the state of the menu

#### Environment

- Angular: 9.1.12
- CDK/Material: 9.2.4
- Browser(s): Chrome
- Operating System (e.g. Windows, macOS, Ubuntu): Windows

**Comment by crisbeto** [comment]
This behavior is based on recommendations from the W3C here: https://www.w3.org/TR/wai-aria-practices-1.1/#menubutton

> When the menu is displayed, the element with role button has aria-expanded set to true. **When the menu is hidden, it is recommended that aria-expanded is not present**. If aria-expanded is specified when the menu is hidden, it is set to false.

**Comment by steven-curran-ub** [comment]
> This behavior is based on recommendations from the W3C here: https://www.w3.org/TR/wai-aria-practices-1.1/#menubutton
> 
> > When the menu is displayed, the element with role button has aria-expanded set to true. **When the menu is hidden, it is recommended that aria-expanded is not present**. If aria-expanded is specified when the menu is hidden, it is set to false.

The new link to the related best-practices for menus is https://www.w3.org/WAI/ARIA/apg/patterns/menu-button/ and it still contains the same suggestion. However, [W3's tutorial on menus](https://www.w3.org/WAI/tutorials/menus/flyout/#fly-out-functionality) (see the "Space Bears" nav button specifically) persists the `aria-expanded` property with a "false" value. Feels like we're getting mixed signals from W3 themselves...

My team is facing questions around this as well, and I stumbled upon this issue. Just posting that updated link and context for others.

It looks like this was assigned at some point, does the team have plans to change the behavior or allow developers to specify whether to persist `aria-expanded` or not? Just curious.
